### PR TITLE
HAAR-884 Provide endpoint to find by username and source - which also saves to db if doesn't already exist

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt [hunk 1]
```diff
@@ -141,6 +141,44 @@ class UserController(private val userService: UserService) {
     val backupVerified: Boolean
   )
 
+  @GetMapping("/api/user")
+  @Operation(
+    summary = "Finds a user based on their username and source and returns their User detail. " +
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:146]: Not convinced on the endpoint name - any suggestions welcome

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:146]: Struggling to think of alternative

**Review by AlTurner-MOJ — APPROVED** [review]
Couple of comments but looks good

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserControllerTest.kt [hunk 2]
```diff
@@ -97,6 +104,96 @@ class UserControllerTest {
     )
   }
 
+  @Test
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserControllerTest.kt:107]: you're probably correct.  Possibly 2 different successes - one for Nomis and one for non-nomis (as need to verify additional data).  Will remove Azure and Delius

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**